Firebase/ios/BonfireTests/PersistedChannelServiceTests.swift

```diff
@@ -0,0 +1,253 @@
+        ]
+
+        XCTAssertEqual(testableChannelsObserver.events, expectedEvents)
+    }
+
+
+    func testThatItReturnsUsersForChannel() {
+        // When
+        channelService.users(forChannel: privateChannel).subscribe(testableUsersObserver)
+        scheduler.start()
+
+        // Then
+        let users = [testUser1, testUser2]
+        let expectedEvents = [
+            next(0, Users(users: users))
+        ]
+        
+        XCTAssertEqual(testableUsersObserver.events, expectedEvents)
+    }
+
+    func testThatItCallsTheRightThingsForCreatingAPublicChannel() {
+        channelService.createPublicChannel(withName: "🐣").subscribe().addDisposableTo(disposeBag)
+
+        let expectedChannel = Channel(name: "🐣", access: .Public)
+        let expectedMethodStack = [
+            MethodCall(identifier: "ChannelsDatabase - writeChannel", arguments: [expectedChannel]),
+            MethodCall(identifier: "ChannelsDatabase - writeChannelToPublicChannelIndex", arguments: [expectedChannel])
+        ]
+
+        XCTAssertEqual(remembrance.callStack, expectedMethodStack)
+    }
+
+    func testThatItCallsTheRightThingsForCreatingAPrivateChannel() {
+        channelService.createPrivateChannel(withName: "🙈", owner: testUser1).subscribe().addDisposableTo(disposeBag)
+
+        let expectedChannel = privateChannel
+        let expectedMethodStack = [
+            MethodCall(identifier: "ChannelsDatabase - addOwnerToPrivateChannel", arguments: [testUser1, expectedChannel]),
+            MethodCall(identifier: "ChannelsDatabase - writeChannel", arguments: [expectedChannel]),
+            MethodCall(identifier: "ChannelsDatabase - addChannelToUserPrivateChannelIndex", arguments: [testUser1, expectedChannel])
+        ]
+
+        XCTAssertEqual(remembrance.callStack, expectedMethodStack)
+    }
+
+    func testThatItCallsTheRightThingsForAddingOwnerToChannel() {
+        channelService.addOwner(testUser1, toPrivateChannel: privateChannel).subscribe().addDisposableTo(disposeBag)
+
+        let expectedChannel = privateChannel
+        let expectedMethodStack = [
+            MethodCall(identifier: "ChannelsDatabase - addOwnerToPrivateChannel", arguments: [testUser1, expectedChannel]),
+            MethodCall(identifier: "ChannelsDatabase - addChannelToUserPrivateChannelIndex", arguments: [testUser1, expectedChannel])
+        ]
+
+        XCTAssertEqual(remembrance.callStack, expectedMethodStack)
+
+    }
+
+
+    func testThatItCallsTheRightThingsForRemovingOwnerFromChannel() {
+        channelService.removeOwner(testUser1, fromPrivateChannel: privateChannel).subscribe().addDisposableTo(disposeBag)
+
+        let expectedChannel = privateChannel
+        let expectedMethodStack = [
+            MethodCall(identifier: "ChannelsDatabase - removeOwnerFromPrivateChannel", arguments: [testUser1, expectedChannel]),
+            MethodCall(identifier: "ChannelsDatabase - removeChannelFromUserPrivateChannelIndex", arguments: [testUser1, expectedChannel])
```

Comment:
I'd prefer this stuff wasn't stringly-typed, but not sure how best to do it in Swift. It's at times like these I miss the dynamism of Objective-C...